### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by aasaru**
* move sections (publishing website, releasing) that are only relevant to few community members to the end of the readme
* fix broken link in table of contents
* turn different configuration list items into links to respective sections
* remove separate chapters "Handling attachment indexing" and include the same info into respective place in documentation (to avoid information being repeated in several places)

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -576,3 +509,76 @@ By default James is configured without LDAP support.
 == Know more about James design
 
 James comes with a https://james.apache.org/documentation.html[Documentation] and https://github.com/linagora/james-project/tree/master/src/adr[Architectural Decision Records].
+
+== Articles for James community
+
+=== How to build and publish the website
```
Comment (chibenwa) [inline README.adoc:493]: I propose to be moving these files to `/docs/modules/community/pages/website.md` and add a link to it in the README.

File: README.adoc [hunk 2]
```diff
@@ -576,3 +509,76 @@ By default James is configured without LDAP support.
...
+
+    $ docker run -v $PWD:/origin -v $PWD/../james-site/content:/destination james/homepage master
+    $ docker run -v $PWD/.m2:/root/.m2 -v $PWD/../james-site/content:/origin -v $PWD/site:/destination james/site master
+
+Then just push the new site:
+
+    $ cd ../james-site
+    $ git push origin asf-site
+
+=== How to release via maven release plugin
```
Comment (chibenwa) [inline README.adoc:545]: I propose to be moving these files to `/docs/modules/community/pages/release.md` and add a link to it in the README.

File: README.adoc [hunk 3]
```diff
@@ -161,18 +161,18 @@ You need a running *rabbitmq* in docker. To achieve this run:
 
 You need a running *s3* compatible objectstorage in docker. To achieve this run:
 
-    $ docker run -d --env 'REMOTE_MANAGEMENT_DISABLE=1' --env 'SCALITY_ACCESS_KEY_ID=accessKey1' --env 'SCALITY_SECRET_ACCESS_KEY=secretKey1' --name=s3 zenko/cloudserver:8.2.6
+    $ docker run -d -p 8080:8000 --env 'REMOTE_MANAGEMENT_DISABLE=1' --env 'SCALITY_ACCESS_KEY_ID=accessKey1' --env 'SCALITY_SECRET_ACCESS_KEY=secretKey1' --name=s3 zenko/cloudserver:8.2.6
```
Comment (chibenwa) [inline README.adoc:164]: Adding back the port binding seems unrelated with the commit title `JAMES-3187 remove separate chapters "Handling attachment indexing"`

**Review by chibenwa — COMMENTED** [review]
Great to have a little lifting on the README!

Thanks for taking some your time doing it.

I think we could be taking this further by moving section of little interest into `/docs` folder (upcoming antora documentation) and linking to it.

Cheers,

Benoit

File: README.adoc (hunk 1 above)
Comment (aasaru) [reply README.adoc]: I think it's a good plan to move sections of little interest away from readme. I created website.adoc (not website.md to keep the pattern of other files) and moved content there. I added a direct link to it from readme. If I understand correctly then once the antora documentation is ready and published the link will somewhat change.

**Review by Arsnael — APPROVED** [review]
Big thanks on this, it is greatly appreciated :)

**Comment by chibenwa** [comment]
Cool! Can we make sure your new pages are references within `docs/modules/community/nav.adoc` ?

(Given that, I agree with the PR too)

**Review by chibenwa — APPROVED** [review]
Nice !

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.

One very last thing: I don't have write access to this repository, hence I can not close this issue. Wouldn't you mind doing it for me? It would avoid me annoying the Apache INFRA team with such simple concerns.

Thanks again,

Cheers,

Benoit
